examples/rosettacode/ethiopian_multiplication.nit

```diff
@@ -0,0 +1,40 @@
+module ethiopian_multiplication
+
+redef class Int
+
+	fun half: Int
+	do
+		return self/2
+	end
+
+	fun double: Int
+	do
+		return self*2
+	end
+end
+
+fun ethiopian(x, y: Int): Int
+   	do
+    	var t = new HashMap[Int, Int]
+    	t[x] = y
+        print "{x}; {t[x]}"
+    	var sum: Int
+    	if x.is_even then 
+    		sum = 0
+    	else
+    		sum = y
+    	end
+
+    	while x>1 do
+    		x = x.half
```

Comment:
Instead of doing `x.half`, you could do `x /= 2`, this would simplify the reading of the code and allow for a removal of the half and double methods in `Int`